src/core/queryObserver.ts

```diff
@@ -68,7 +68,7 @@ export class QueryObserver<
     TQueryKey
   >
   private previousQueryResult?: QueryObserverResult<TData, TError>
-  private previousSelectError: Error | null
+  private previousSelectError: TError | null
```

Comment:
this change seems unrelated. Why is it necessary please?

Reply:
These are type errors in the latest version of Typescript. I can fix in separate PR if you prefer.

Reply:
yes lets please do this in a different branch, where we also update our TS version to the latest (4.5 as of a couple of minutes ago)